src/llama-arch.h

```diff
@@ -156,6 +157,13 @@ enum llm_kv {
     LLM_KV_ATTENTION_VALUE_LENGTH_MLA,
     LLM_KV_ATTENTION_LAYER_INDICES,
 
+    // Falcon-H1 specific
+    LLM_KV_ATTN_HEAD_DIM,
+    LLM_KV_SSM_HEAD_DIM,
+    LLM_KV_MAMBA_D_SSM,
+    LLM_KV_N_LAYER,
+    LLM_KV_FALCON_H1_MAMBA_RMS_NORM,
```

Comment:
Can this be simply `LLM_KV_MAMBA_RMS_NORM`? If there isn't anything very specific for Falcon H1, it's better to keep the names generic.